### PR TITLE
DRILL-8225: Update LogParser and Yauaa to support User-Agent Client Hints

### PR DESCRIPTION
**Pull request by nielsbasjes**
# [DRILL-8225](https://issues.apache.org/jira/browse/DRILL-8225): Update LogParser and Yauaa to support User-Agent Client Hints

## Description

In order to support logging, parsing and analyzing the User-Agent Client Hints it was needed to update the LogParser to fix a parsing bug and to update Yauaa to support actually analyzing them.

See https://yauaa.basjes.nl/using/clienthints/

## Documentation
The change includes the updated Readme for the UDFs.

## Testing
The change includes the original tests and additional tests for the new functionality.
For some of the existing tests the readability of the test has been improved.

For specific edge cases (like passing a null value and empty string) the parse results have changed.

### CODE REVIEW

File: contrib/udfs/src/test/java/org/apache/drill/exec/udfs/TestUserAgentFunctions.java [hunk 1]
```diff
@@ -168,4 +241,185 @@ public void testEmptyUAString() throws Exception {
...
+      )
+      .go();
+  }
+
+  /**
+   * Converts a more readable list of keys and values into what the ClusterTest supports.
+   * @param strings List of  ["key", "value"]
+   * @return A Map of the same keys and values that is in the right format.
+   */
+  private Map<String, Object> expectations(String... strings) {
```
Comment (cgivre) [inline contrib/udfs/src/test/java/org/apache/drill/exec/udfs/TestUserAgentFunctions.java:413]: This is a nice function.

**Review by cgivre — APPROVED** [review]
LGTM +1
Thanks for the PR!
